programs/MouseDemo.hs

```diff
@@ -51,6 +56,18 @@ buttonLayer st =
                padLeftRight (if wasClicked then 2 else 3) $
                str (if wasClicked then "<" <> label <> ">" else label)
 
+editLayer :: St -> Widget Name
+editLayer st =
+  B.border $
+  C.hCenterLayer $
+  vLimit 8 $
+  -- n.b. if clickable and viewport are inverted here, click event
+  -- coordinates will only identify the viewable range, not the actual
+  -- editor widget coordinates.
```

Comment:
As followup, using the editor's internal viewport, the clickable row reporting is always 0-7, regardless of the scrolling from the viewport (which matches the behavior I observed here as well).  That's not necessarily optimal for determining which line of text was clicked on.. that's where the getEventWidgetLocation helper function could be useful (which was supposed to be a separate pull request; git surprised me, and I cancelled this pull request to submit two independent cleaner requests).